cpp/src/arrow/io/memory.h

```diff
@@ -161,6 +161,8 @@ class ARROW_EXPORT BufferReader : public RandomAccessFile {
   std::shared_ptr<Buffer> buffer() const { return buffer_; }
 
  protected:
+  inline Status CheckClosed() const;
```

Comment:
does `inline` have any effect here (curious)?

Reply:
I'm hoping that it incites the compiler to inline the implementation into its callers... @fsaintjacques am I right?

Reply:
Not sure if `inline` will have the desired effect, but if it is marked `inline` then this function should be private, not protected: if `BufferReader` is subclassed in another translation unit and the subclass calls `CheckClosed` then we'll get a linker error when the inline function isn't defined